src/notification/NotificationController.ts

```diff
@@ -1,48 +1,52 @@
 import BaseController, { BaseConfig, BaseState } from '../BaseController';
 
-interface action {
-  actionText: string;
-  actionFunction: VoidFunction;
-}
-
 interface viewedNotification {
   [id: number]: boolean;
 }
 
-export interface Notification{
+interface Notification {
   id: number;
   title: string;
   description: string;
   date: string;
   image?: string;
-  action?: Partial<action>;
   isShown?: boolean;
+  actionText?: string;
+}
+
+/**
+  * A map of notification ids to Notification objects
+  */
+interface NotificationMap {
+  [id: number]: Notification;
 }
 
 /**
  * NotitificationConfig will hold the notifications from JSON file read
  * from `metamask-extension`
  */
 export interface NotificationConfig extends BaseConfig{
-  allNotifications: Notification[];
+  allNotifications: NotificationMap | undefined;
```

Comment:
Why ` | undefined`? I wouldn't expect this controller to be constructed in the first place if there are no notifications

Reply:
Removed in 371db50e (This was left over from when I was anticipating a possible need to only add notifications at some point after the controller had been constructed, via another public method. But it turns out we won't be doing that.)